script.js

```diff
@@ -0,0 +1,54 @@
+const parse = require('gemini-to-html/parse')
+const render = require('gemini-to-html/render')
+
+if (document.body.children.length === 1 &&
+document.getElementsByTagName('pre')[0] !== undefined) {
+  fetch(location.href).then(response => {
+    if (response.headers.get('Content-Type') === 'text/gemini') {
+      response.text().then(text => {
+        let parsed = parse(text)
+        let rendered = render(parsed)
+
+        let title = parsed.find(element => element.type === 'header').content || 'Gemini Document' // Find relevant page content here?
```

Comment:
Would you mind reusing the same code as the protocol handler for getting the title? https://github.com/RangerMauve/agregore-browser/blob/master/app/protocols/gemini-protocol.js#L59

I think this has a bug where if there are no elements with `type: 'header'` it'll throw an error because it'll try to get `.content` of `undefined`.

Reply:
Yup, I recalled that there was an elegant way already written but couldn't think of where... must have been late.

Ah yes! I was playing with another way of doing it for a while and then decided, this is stupid? Why don't I do it this way?

I recall now that there was a reason I was doing it the other way haha, thanks.